R/autosyn.R

```diff
@@ -472,3 +475,210 @@ flywire_ntplot3d <- function(x, nts=c("gaba", "acetylcholine", "glutamate",
+#' neurons.syns = flywire_neurons_add_synapses(neurons, transmitters = TRUE)
+#' neurons.syns[,]
+#'
+#' # Plot in 3D
+#' library(catmaid)
+#' nopen3d()
+#' plot3d(neurons.syns, WithConnectors = TRUE)
+#'
+#' # Axon-dendrite split
+#' library(hemibrainr)
+#' neurons.flow = flow_centrality(neurons.syns,
+#'   polypre = TRUE,
+#'   mode = "centrifugal")
+#' clear3d()
+#' plot3d_split(neurons.flow, WithConnectors = TRUE,
+#' transmitter = TRUE,
+#' radius = 1000, soma = 4000)
+#' }
+#' }
+flywire_neurons_add_synapses <- function(x,
+                                         connectors = NULL,
+                                         cloudvolume.url=NULL, # fafbseg.cloudvolume.url="graphene://https://prodv1.flywire-daf.com/segmentation/1.0/fly_v31"
+                                         method=c("auto", "spine", "sqlite"),
+                                         remove_autapses = TRUE,
+                                         cleft.threshold = 0,
+                                         Verbose=TRUE,
+                                         transmitters=TRUE,
+                                         local = NULL, # "/Volumes/nnautilus/projects/JanFunke"
+                                         ...) UseMethod("flywire_neurons_add_synapses")
+
+#' @export
+#' @rdname flywire_neurons_add_synapses
+flywire_neurons_add_synapses.neuron <- function(x,
+                                                connectors = NULL,
+                                                cloudvolume.url=NULL,
+                                                method=c("auto", "spine", "sqlite"),
+                                                remove_autapses = TRUE,
+                                                cleft.threshold = 0,
+                                                Verbose=TRUE,
+                                                transmitters=TRUE,
+                                                local = NULL,
+                                                ...){
+  method = match.arg(method)
+  rootid = x$flywire.id
+  if(is.null(connectors)){
+    synapses = flywire_partners(rootid = rootid,
+                                partners = "both",
+                                roots=TRUE,
+                                details=TRUE,
+                                cloudvolume.url = cloudvolume.url,
+                                method=method,
+                                Verbose=Verbose,
+                                local = local,
+                                ...)
+    if(is.null(synapses) || nrow(synapses)==0){
+      class(x) = union(c("flywireneuron", "catmaidneuron"), class(x))
+      return(x)
+    }
+    if(remove_autapses) {
+      synapses=synapses[synapses$post_id!=synapses$pre_id,,drop=FALSE]
+    }
+    # Add synapses
+    synapses %>%
+      dplyr::filter(.data$cleft_scores >= cleft.threshold) %>%
+      dplyr::mutate(x = ifelse(rootid==.data$pre_svid, .data$pre_x, .data$post_x)) %>%
```

Comment:
@alexanderbates `rootid` will never == a supervoxel id , so this means all positions will be `pre_x` never `post_x`. Shouldn't you use `prepost` for this logic?

Reply:
Please can you fix!

Reply:
addressed